### PR TITLE
#66 Add cache for local file import.

### LINKED ISSUE #66: add cache to avoid multiple identified imports
### Current arrai Version
v0.1.0 4add794

### Use-cases
Add cache to avoid multiple identified imports. e.g.
```
...
///foo/bar.math.pi
...
///foo/bar.math.pi
...
```
only import `///foo/bar.math` once

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #66  .

Changes proposed in this pull request:
- Add cache for import behavior
- 

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/import.go [hunk 1]
```diff
@@ -147,8 +148,13 @@ func importURL(url string) (rel.Value, error) {
 		if err != nil {
 			return nil, err
 		}
-
-		return bytesValue("", data), nil
+		if exists, val := cache.exists(url); exists {
+			return val, nil
+		}
+		// Pass space as file path as it is remote url and can't be found in local env
+		val := bytesValue("", data)
```
Comment (ChloePlanet) [inline syntax/import.go]: Sorry I just read the master branch code and realized we should use `NoPath` here instead of empty string.

Comment (ChloePlanet) [reply syntax/import.go]:
```suggestion
		val := bytesValue(NoPath, data)
```

File: syntax/import.go [hunk 2]
```diff
@@ -166,6 +172,13 @@ func fileValue(filename string) (rel.Value, error) {
 }
 
 func bytesValue(filename string, data []byte) rel.Value {
+	// maybe filename is "", so add check first
+	if len(strings.TrimSpace(filename)) > 0 {
```
Comment (ChloePlanet) [inline syntax/import.go]:
```suggestion
	if filename != NoPath {
```

File: syntax/import.go [hunk 3]
```diff
@@ -175,5 +188,9 @@ func bytesValue(filename string, data []byte) rel.Value {
 		panic(err)
 	}
 
+	// maybe filename is "", so add check first
+	if len(strings.TrimSpace(filename)) > 0 {
```
Comment (ChloePlanet) [inline syntax/import.go]:
```suggestion
	if filename != NoPath {
```

File: syntax/import.go (hunk 2 above)
Comment (ChloePlanet) [reply syntax/import.go]: Same as the first comment.

File: syntax/import.go (hunk 1 above)
Comment (ericzhang6222) [reply syntax/import.go]: Updated.

File: syntax/import.go (hunk 2 above)
Comment (ericzhang6222) [reply syntax/import.go]: Fixed.

File: syntax/import.go (hunk 3 above)
Comment (ericzhang6222) [reply syntax/import.go]: Fixed.

**Pull request merged**